rotkehlchen/tests/unit/test_inquirer.py

```diff
@@ -502,3 +504,24 @@ def mock_coingecko_return(url, *args, **kwargs):  # pylint: disable=unused-argum
         inquirer.find_usd_price(A_ETH, ignore_cache=True)
         # Check that the last price query contacted coingecko api
         assert coingecko_api_calls == 2
+
+
+@pytest.mark.parametrize('should_mock_current_price_queries', [False])
+def test_punishing_of_oracles_works(inquirer):
+    coingecko_patch = patch.object(inquirer._coingecko.session, 'get', side_effect=requests.exceptions.RequestException('An unexpected error occurred!'))  # noqa: E501
+    inquirer.set_oracles_order(oracles=[CurrentPriceOracle.COINGECKO, CurrentPriceOracle.DEFILLAMA])  # noqa: E501
```

Comment:
hmm no that's not what I meant. You are not mockking deffillama here, which means you are making 6 calls to defillama. Don't do it. Mockthat too. There is no reason to do remote queries **at all** in this test